File: my_simple_robot_nodes/scripts/simple_robot_receiver.py
#! /usr/bin/env python3
"""
    Simple ROS subscriber node
    
    Receives and prints /robot_news String Message

    Author: Jari Honkanen
"""

import rospy
from std_msgs.msg import String


def callback_receive_message(msg):
    """ Callback function for received message """
    rospy.loginfo("[INFO] Message Received: ")
    rospy.loginfo(msg)


if __name__ == "__main__":

    rospy.init_node("simple_robot_receiver")

    rospy.loginfo("[INFO] simple_robot_receiver Node started")    

    # Create subscriber:  message name, message type, callback function
    sub = rospy.Subscriber("/robot_news", String, callback_receive_message)

    # Keep script running with callback active until node stopped
    rospy.spin()

File: my_simple_robot_nodes/src/simple_robot_receiver.cpp
/*-------------------------------------------------------------------*\
  NAME
    simple_robot_receiver.cpp

  DESCRIPTION
    Simple ROS subscriber node 
    Receives and prints /robot_news String Message

  AUTHOR
    Jari Honkanen
\*-------------------------------------------------------------------*/

#include <ros/ros.h>
#include <std_msgs/String.h>


// Callback function for receiving messages
void callback_receive_message(const std_msgs::String &msg)
{
    ROS_INFO("[INFO] Message Received: '%s'", msg.data.c_str());
}


int main(int argc, char **argv)
{
    // Initialize node
    ros::init(argc, argv, "simple_robot_receiver");

    ROS_INFO("[INFO] simple_robot_receiver started");

    // Instantiate node handle
    ros::NodeHandle nh;

    // topic name, queue size, callback function
    ros::Subscriber sub = nh.subscribe("/robot_news", 100, callback_receive_message);  

    // Keep node running with callback active until shutdown
    ros::spin();

}